six/modules/c++/six.sicd/tests/test_add_additional_des.cpp

```diff
@@ -0,0 +1,252 @@
+/* =========================================================================
+* This file is part of six.sidd-c++
+* =========================================================================
+*
+* (C) Copyright 2004 - 2016, MDA Information Systems LLC
+*
+* six-c++ is free software; you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; If not,
+* see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <six/NITFWriteControl.h>
+#include <six/Types.h>
+#include <six/XMLControlFactory.h>
+#include <six/sicd/Utilities.h>
+
+
```

Comment:
You use an unnamed namespace to basically hide functions in a cpp so that there's no way someone could link and use it.  So helper functions that aren't intended to be used elsewhere.  In an actual application cpp where nothing else is going to link in, it's just a style/consistency thing.